providers/base/tests/test_data/kdrp-kdrp-k4500-gadget.yaml

```diff
@@ -0,0 +1,104 @@
+  - plug: qphgGdZKBvUcYyW19xDkcASkc7sFby6P:tpm
+    slot: system:tpm
+  ## connect avnet-avt-iiotg20-kernel:hardware-observe
+  - plug: qphgGdZKBvUcYyW19xDkcASkc7sFby6P:hardware-observe
+    slot: system:hardware-observe
+  # Connect the KDRP gadget snap serial ports to the kdrp-optimus-application snap
+  - plug: 8on7qeWMy6YnffVrluPrSd9zyDxctDBD:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:bt-serial
+  - plug: 8on7qeWMy6YnffVrluPrSd9zyDxctDBD:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-brew0
+  - plug: 8on7qeWMy6YnffVrluPrSd9zyDxctDBD:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-brew1
+  - plug: 8on7qeWMy6YnffVrluPrSd9zyDxctDBD:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-mmodem0
+  - plug: 8on7qeWMy6YnffVrluPrSd9zyDxctDBD:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-nmodem0
+  - plug: 8on7qeWMy6YnffVrluPrSd9zyDxctDBD:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-powder
+  - plug: 8on7qeWMy6YnffVrluPrSd9zyDxctDBD:i2c
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:i2c-1
+  - plug: 8on7qeWMy6YnffVrluPrSd9zyDxctDBD:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-honeywell-1900
+  - plug: 8on7qeWMy6YnffVrluPrSd9zyDxctDBD:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-honeywell-1900c
+  - plug: 8on7qeWMy6YnffVrluPrSd9zyDxctDBD:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-tmodem0
+  - plug: 8on7qeWMy6YnffVrluPrSd9zyDxctDBD:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-tmodem1
+  # Connect the KDRP gadget snap serial ports to the kdrp-application snap
+  - plug: y78spNO0khAv2JK2f1bbCtOyCcZ5N3st:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:bt-serial
+  - plug: y78spNO0khAv2JK2f1bbCtOyCcZ5N3st:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-brew0
+  - plug: y78spNO0khAv2JK2f1bbCtOyCcZ5N3st:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-brew1
+  - plug: y78spNO0khAv2JK2f1bbCtOyCcZ5N3st:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-mmodem0
+  - plug: y78spNO0khAv2JK2f1bbCtOyCcZ5N3st:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-nmodem0
+  - plug: y78spNO0khAv2JK2f1bbCtOyCcZ5N3st:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-powder
+  - plug: y78spNO0khAv2JK2f1bbCtOyCcZ5N3st:i2c
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:i2c-1
+  - plug: y78spNO0khAv2JK2f1bbCtOyCcZ5N3st:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-honeywell-1900
+  - plug: y78spNO0khAv2JK2f1bbCtOyCcZ5N3st:serial-port
+    slot: AhXK9WxxcNAY6JP9XLejXAzUV8Y3tVyw:serial-port-honeywell-1900c
+
+
+defaults:
+  # system retention settings
+  system:
+    watchdog:
+      runtime-timeout: 15s
+      shutdown-timeout: 15s
+    #pi-config:
+    #  hdmi-force-hotplug: 1
+    journal:
+      persistent: true
+    service:
+      console-conf.disable: true
```

Comment:
```suggestion
```

This part of the test_data is not relevant for the test you're running, so you can remove it.